test/skimNtuple_HHbtag.cpp

```diff
@@ -3200,35 +3202,35 @@ int main (int argc, char** argv)
 
-				  double Eff_Data_mu_up   = passSingle * SFL_Data_up   - passCross * passSingle * std::min(SFl_Data_up  , SFL_Data_up)   * SFtau_Data + passCross * SFl_Data_up   * SFtau_Data;
-				  double Eff_MC_mu_up     = passSingle * SFL_MC_up     - passCross * passSingle * std::min(SFl_MC_up    , SFL_MC_up)     * SFtau_MC   + passCross * SFl_MC_up     * SFtau_MC;
-				  double Eff_Data_mu_down = passSingle * SFL_Data_down - passCross * passSingle * std::min(SFl_Data_down, SFL_Data_down) * SFtau_Data + passCross * SFl_Data_down * SFtau_Data;
-				  double Eff_MC_mu_down   = passSingle * SFL_MC_down   - passCross * passSingle * std::min(SFl_MC_down  , SFL_MC_down)   * SFtau_MC   + passCross * SFl_MC_down   * SFtau_MC;
-
-				  // for each DM, fill a trigSF branch with the up/down values if the tauh has the corresponding DM, otherwise fill with nominal trigSF value
-				  vector <double> SFtau_Data_up   (N_tauhDM, SFtau_Data);
-				  vector <double> SFtau_Data_down (N_tauhDM, SFtau_Data);
-				  vector <double> SFtau_MC_up     (N_tauhDM, SFtau_MC);
-				  vector <double> SFtau_MC_down   (N_tauhDM, SFtau_MC);
-				  vector <double> Eff_Data_up     (N_tauhDM, Eff_Data);
-				  vector <double> Eff_Data_down   (N_tauhDM, Eff_Data);
-				  vector <double> Eff_MC_up       (N_tauhDM, Eff_MC);
-				  vector <double> Eff_MC_down     (N_tauhDM, Eff_MC);
-
-				  for (int idm  = 0; idm < N_tauhDM; idm ++)
-					{
-					  if (isthisDM_second[idm])
-						{
-						  SFtau_Data_up[idm]   = tauTrgSF_mutau->getEfficiencyData(tlv_secondLepton.Pt(), DM2, 1);
-						  SFtau_Data_down[idm] = tauTrgSF_mutau->getEfficiencyData(tlv_secondLepton.Pt(), DM2, -1);
-						  SFtau_MC_up[idm]     = tauTrgSF_mutau->getEfficiencyMC  (tlv_secondLepton.Pt(), DM2, 1);
-						  SFtau_MC_down[idm]   = tauTrgSF_mutau->getEfficiencyMC  (tlv_secondLepton.Pt(), DM2, -1);
+				  // Compute uncertainties (separate for SingleMuon and mu leg of cross trigger)
+				  double Err_Data_SL_mu = passSingle * SFL_Data_Err - passCross * passSingle * (SFl_Data > SFL_Data) * SFL_Data_Err * SFtau_Data;
+				  double Err_MC_SL_mu   = passSingle * SFL_MC_Err   - passCross * passSingle * (SFl_MC   > SFL_MC)   * SFL_MC_Err   * SFtau_MC;
+
+				  double trigSF_SL_mu_err = muTrgSF->get_ScaleFactorError(Eff_Data, Eff_MC, Err_Data_SL_mu, Err_MC_SL_mu);
+
+				  double Err_Data_cross_mu = - passCross * passSingle * (SFl_Data < SFL_Data) * SFl_Data_Err * SFtau_Data + passCross * SFl_Data_Err * SFtau_Data;
+				  double Err_MC_cross_mu   = - passCross * passSingle * (SFl_MC   < SFL_MC)   * SFl_MC_Err   * SFtau_MC   + passCross * SFl_MC_Err   * SFtau_MC;
+
+				  double trigSF_cross_mu_err = muTrgSF->get_ScaleFactorError(Eff_Data, Eff_MC, Err_Data_cross_mu, Err_MC_cross_mu);
+
+				  double Err_Data_mu = Err_Data_SL_mu + Err_Data_cross_mu;
+				  double Err_MC_mu   = Err_MC_SL_mu   + Err_MC_cross_mu;
+
+				  double trigSF_mu_err   = muTrgSF->get_ScaleFactorError(Eff_Data, Eff_MC, Err_Data_mu, Err_MC_mu);
```

Comment:
I'm not sure I understand how this is different from the `std::min` implementation we had? Were there events in which the "nominal" made use e.g. of the singleLep efficiency in the crossterm while the up/down used the crosstrigger efficiency? in what situation?

Reply:
it's possible it does the exact same, I just wanted to make sure, because it seemed weird to use the one trigger efficiency for the nominal and then the other for one of the uncertainties